container/libcontainer/helpers.go

```diff
@@ -224,16 +221,14 @@ func scanTcpStats(tcpStatsFile string) (info.TcpStat, error) {
 	scanner.Split(bufio.ScanLines)
 
 	for scanner.Scan() {
-
 		line := scanner.Text()
-		//skip header
-		matched := tcpStatLineRE.MatchString(line)
-
-		if matched {
-			state := strings.Fields(line)
-			//#file header tcp state is the 4 filed:
-			//sl local_address rem_address st tx_queue rx_queue tr tm->when retrnsmt  uid timeout inode
-			tcpStateMap[state[3]]++
+
+		state := strings.Fields(line)
+		//#file header tcp state is the 4th field:
+		//sl local_address rem_address st tx_queue rx_queue tr tm->when retrnsmt  uid timeout inode
+		tcpState := state[3]
+		if _, ok := tcpStateMap[tcpState]; ok {
```

Comment:
If this is not OK it means you encountered an unexpected state. I think it's worth logging an error or warning if that happens (although that could generate a LOT of log spam if the file is corrupted, so it might be good to rate limit to a single log entry per call)